_episodes/04-networks-are-like-onions.md

```diff
@@ -0,0 +1,544 @@
+
+Let's put it into practice. We compose a Convolutional network with two convolutional layers and two pooling layers.
+
+
+~~~
+inputs = keras.Input(shape=train_images.shape[1:])
+x = keras.layers.Conv2D(32, (3, 3), activation='relu')(inputs)
+x = keras.layers.MaxPooling2D((2, 2))(x)
+x = keras.layers.Conv2D(32, (3, 3), activation='relu')(x)
+x = keras.layers.MaxPooling2D((2, 2))(x)
+x = keras.layers.Flatten()(x)
+x = keras.layers.Dense(32, activation='relu')(x)
+outputs = keras.layers.Dense(10)(x)
+
+model = keras.Model(inputs=inputs, outputs=outputs, name="cifar_model_small")
+
+model.summary()
+~~~
+{: .language-python}
+~~~
+    Model: "cifar_model_small"
+    _________________________________________________________________
+    Layer (type)                 Output Shape              Param #
+    =================================================================
+    input_6 (InputLayer)         [(None, 32, 32, 3)]       0
+    _________________________________________________________________
+    conv2d_3 (Conv2D)            (None, 30, 30, 32)        896
+    _________________________________________________________________
+    max_pooling2d (MaxPooling2D) (None, 15, 15, 32)        0
+    _________________________________________________________________
+    conv2d_4 (Conv2D)            (None, 13, 13, 32)        9248
+    _________________________________________________________________
+    max_pooling2d_1 (MaxPooling2 (None, 6, 6, 32)          0
+    _________________________________________________________________
+    flatten (Flatten)            (None, 1152)              0
+    _________________________________________________________________
+    dense_1 (Dense)              (None, 32)                36896
+    _________________________________________________________________
+    dense_2 (Dense)              (None, 10)                330
+    =================================================================
+    Total params: 47,370
+    Trainable params: 47,370
+    Non-trainable params: 0
+    _________________________________________________________________
+~~~
+{: .output}
+
+We compile the model and train it on our training data for 10 epochs
+
+~~~
+model.compile(optimizer='adam',
+              loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+              metrics=['accuracy'])
+
+history = model.fit(train_images, train_labels, epochs=10,
+                    validation_data=(test_images, test_labels))
+~~~
+{: .language-python}
+
+
+We can plot the training process using the history:
+
+~~~
+history_df = pd.DataFrame.from_dict(history.history)
+sns.lineplot(data=history_df[['accuracy', 'val_accuracy']])
+~~~
+{: .language-python}
+![Output of plotting sample](../fig/04_training_history_1.png)
+~~~
+sns.lineplot(data=history_df[['loss', 'val_loss']])
+~~~
+{: .language-python}
+![Output of plotting sample](../fig/04_training_history_loss_1.png)
+
```

Comment:
This is the first time pandas and seaborn are used in this lesson, so I suggest we add an import statement for them.